### PR TITLE
Issue/unql 14/create api for compress equal subsequential messages

### PR DESCRIPTION
**Pull request by MatteoAns**

### CODE REVIEW

File: lib/src/ConsoleWriter.h [hunk 1]
```diff
@@ -18,7 +18,11 @@ class ConsoleWriter: public LogWriter
     UNQL::ConsoleColorScheme m_colorScheme;
     static QMutex m_consoleMux;
 
-    QString getColorCode(const LogMessage& i_message);
+    QString getColorCode(const UNQL::LogMessagePriorityType &i_level);
+
+    void writeCompressedMessages();
```
Comment (alienpenguin) [inline lib/src/ConsoleWriter.h]: these does not exist anymore right?

File: lib/src/LogMessage.cpp [hunk 2]
```diff
@@ -11,17 +11,62 @@
...
-      m_tstamp(aTstamp)
+      m_initTstamp(aInitTstamp),
+      m_endTstamp(aEndTstamp)
+
 {
+    (repetitions < 1) ? (m_repetitions = 1) : (m_repetitions = repetitions);
 }
 
 
 QString
```
Comment (alienpenguin) [inline lib/src/LogMessage.cpp:32]: missing documentation

File: lib/src/LogMessage.cpp [hunk 3]
```diff
@@ -11,17 +11,62 @@
...
+/*!
+ * \brief LogMessage::getTstampString is a static function that will return a string with current timestamp using the default timeformat
+ * \return a string containing current timestamp
+ */
+QString LogMessage::getCurrentTstampString()
+{
+    return QDateTime::currentDateTime().toString(DEF_UNQL_TSTAMP_FMT);
+}
+
+QString LogMessage::repeatedMessage() const
```
Comment (alienpenguin) [inline lib/src/LogMessage.cpp:56]: missing documentation

File: lib/src/LogMessage.cpp [hunk 4]
```diff
@@ -11,17 +11,62 @@
...
+    msg.reserve(4*2 + (m_formatting.spacingSize() * 7) + m_initTstamp.size() + m_endTstamp.size()
+                + m_loggerName.size() + m_msg.size() + priolev.size() + n.size());
+
+    msg = sc + m_initTstamp + spc + "-" + spc + m_endTstamp + ec + spc
+          + sc + m_loggerName + ec + spc + sc + priolev + ec + spc + m_msg + spc + sc + n + ec;
+
+    return msg;
+}
+
+QString LogMessage::singleMessage() const
```
Comment (alienpenguin) [inline lib/src/LogMessage.cpp:85]: missing documentation

File: lib/src/LogWriter.cpp [hunk 5]
```diff
@@ -271,3 +284,64 @@ LogWriter::pauseLogging(bool status)
     m_logIsPaused = status;
 }
  
+void
+LogWriter::compressMessages()
```
Comment (alienpenguin) [inline lib/src/LogWriter.cpp:294]: missing documentation

Comment (alienpenguin) [reply lib/src/LogWriter.cpp:294]: also, can you please make a simple test to measure how much time we spend to log 10000 messagees with old uniqlogger and with the new one with the compress feature on?
ald so to log 10k messages with only groups of 3-4 messages to be crompressed?

File: lib/src/ConsoleWriter.h (hunk 1 above)
Comment (MatteoAns) [reply lib/src/ConsoleWriter.h]: you are right

**Review by alienpenguin — APPROVED** [review]

**Pull request merged**